### ISSUE TITLE
Auto scan + auto connect to known device

### ISSUE THREAD

**Issue by beeradmoore** [OnewheelCommunityEdition/OWCE_App#47]
**Is your feature request related to a problem? Please describe.**
Not a problem.

**Describe the solution you'd like**
When you open the app it should auto scan for bluetooth devices if it has bluetooth permission. Additionally if a device is found and has some "auto connect" flag set it should auto connect to the first known auto connect board.

**Comment by ocornu** [comment]
Super useful feature, much needed. :-D
Building on this: you still need the ability to connect to other boards. Which means, app should auto-connect to your favorite board if it's there, but if you manually disconnect, it should not auto-reconnect to it.
Which brings some interesting point: we don't need to have a "favorite" board, we just want to auto-reconnect to the last board manually connected to until user manually disconnects. No need for a favorite board(s) U.I. :-)

**Comment by beeradmoore** [comment]
I like the idea of auto-connect to multiple boards in order that they were last used.  I totally would have gone for favourites board setting.

Should auto-connect just be the default behaviour for all boards?
And if so, if a user manually disconnects are we essentially just removing that board from the list of auto-connectable boards? 

Would also need to make sure the users are not manually disconnecting at the end of the ride. If needed could add an alert
"Are you sure you want to disconnect?
Yes
Yes, and don't auto connect until I connect again
Cancel"

Unsure if we'd also need an app setting of "auto-connect to last boards", unsure if its something someone would want to turn off completely.

**Comment by ocornu** [comment]
I believe most of this is overkill.
As I see it, what we need simply is a persistent link to a board beyond BLE
disconnections (suppose you walk away from your board to pee behind a bush
for example) and use/charge cycles. Basically, once you connect to a board,
OWCE will connect back to it if it's there. If not, or if you manually
disconnected, then it'll wait for your input.
Maintaining an ordered list won't be as intuitive and may even be
error-prone: for example, suppose your second-to-last board shows up first
during scanning for some reason, even though the last board used also is
around...
With the persistence approach, implemented correctly, people shouldn't even
notice it's there (except for old users who will surely appreciate it!). If
you think about it, that's what a user wants when he connects to a board
--even if he does not realize it. 😊
Thus, done right, there should not be a need for a setting either, although
it can always be added later if a relevant use case shows up.

**Comment by beeradmoore** [comment]
There are two different things there I think.

Auto-connect: for when you open the all from fresh launch
Auto-reconnect: for when you go out of range and come back without restarting or leaving the app.

My above was all about the first one, but both should exist.

**Comment by ocornu** [comment]
I don't believe this distinction makes sense… Do you have a use-case for it?
For starters, at least on Android, people rarely close the app: they just hit "back" or "menu", which put it in the background until you "launch" it again: the app "fresh launch" paradigm does not correlate with user behavior. Secondly, even during normal use, i believe it's not uncommon to have the app reset while you were checking another app (say, a mapping software, answering a phone call, taking a photo/video): you never closed it, yet when you get back to it it's as if you were launching it (it scans again, you have to tap the only board available and the very same one you were connected to moments ago, again…).
Auto-connecting, and if necessary auto-reconnecting to the last board connected to (if it's there) takes care of all use-cases i can think of. I truly believe that's the right starting point. If using it we find something more complex is required, why not? Until then…

**Comment by beeradmoore** [comment]
People close apps all the time regardless of iOS or Android. The OS also closes apps all the time. I don't think people are closing the app mid ride, but some people have intermittent bluetooth issues and the phone loses connection with the board. This is where auto-reconnect is also useful.

Auto-connect is useful for when you grab your board and are about to go start the ride. But I do see now that it isn't just from app closed state, but from app going to foreground while no board is connected should trigger the same connection flow. Auto-scan should do the same thing (well, its kinda needed for auto-connect anyways)
